src/main/java/cz/metacentrum/perun/googlegroupconnector/GoogleGroupsServiceImpl.java

```diff
@@ -800,4 +826,286 @@ private void deleteMember(String groupName, String memberId) throws GoogleGroups
 		}
 	}
 
+
+	/**
+	 * New class representing tuple of team drive and its users from CSV file.
+	 */
+	public class DriveWithMembers {
+		private TeamDrive teamDrive;
+		private List<User> driveUsers = new ArrayList<>();
+
+		DriveWithMembers(TeamDrive teamDrive, List<User> driveUsers) {
+			this.teamDrive = teamDrive;
+			this.driveUsers = driveUsers;
+		}
+
+		public TeamDrive getTeamDrive() {
+			return teamDrive;
+		}
+
+		public void setTeamDrive(TeamDrive teamDrive) {
+			this.teamDrive = teamDrive;
+		}
+
+		public List<User> getDriveUsers() {
+			return driveUsers;
+		}
+
+		public void setDriveUsers(List<User> driveUsers) {
+			this.driveUsers = driveUsers;
+		}
+	}
+
+	@Override
+	public List<DriveWithMembers> parseTeamDrivesFile(File teamDriveFile) {
+
+		List<DriveWithMembers> result = new ArrayList<>();
+		FileReader fileReader = null;
+
+		try {
+
+			fileReader = new FileReader(teamDriveFile);
+
+			char separator = ';';
+			CSVReader reader = new CSVReader(fileReader, separator);
+
+			List<String[]> lines = reader.readAll();
+			if (lines != null && !lines.isEmpty()) {
+
+				for (String[] line : lines) {
+
+					if (line.length < 2) {
+						log.error("TeamDrive file contains row with less than 2 columns: {}", line);
+						throw new IllegalArgumentException("TeamDrive file contains row with less than 3 columns:" + line[0]);
+					}
+
+					TeamDrive teamDriveResult = new TeamDrive();
+					List<User> userListResult = new ArrayList<>();
+
+					teamDriveResult.setName(line[0].toLowerCase());
+					List<String> membersEmail = Arrays.asList(line[1].split(","));
+
+					for (String userMail : membersEmail) {
+						userMail = userMail.replaceAll("\\s+", "");
+						User user = new User();
+						user.setPrimaryEmail(userMail);
```

Comment:
Perun can provide Google with either ``primaryEmail`` od ``ID`` of user (based on what is available). Can Drive API work with both when inserting permissions for user on Drive ? If yes, please use property based on settings, like when we insert members to the group. If it requires only primaryEmail, then we should discuss it / test it, because CESNET instance provides only IDs of google accounts and not email address.

Reply:
As i was looking into documentation, permission has field only for email of specific user. It looks like it will be necessary to test it.